planner/core/resolve_indices.go

```diff
@@ -632,13 +635,60 @@ func (p *baseSchemaProducer) ResolveIndices() (err error) {
 	return
 }
 
+// ResolveIndices implements Plan interface.
+func (p *basePhysicalJoin) ResolveIndices() (err error) {
+	for _, child := range p.children {
+		err = child.ResolveIndices()
+		if err != nil {
+			return err
+		}
+	}
+	switch p.JoinType {
+	case RightOuterJoin:
+		p.AlignNullFlag(p.children[1].Schema())
+		p.EnsureNullFlag(p.schema, 0, len(p.children[0].Schema().Columns))
+	case LeftOuterJoin:
+		p.AlignNullFlag(p.children[0].Schema())
+		p.EnsureNullFlag(p.schema, len(p.children[0].Schema().Columns), len(p.schema.Columns))
+	default:
```

Comment:
That is not accurate: For full outer join, all the columns should be Nullable. For now we are only concern about TiFlash, So select all the join types that TiFLash supports.

Reply:
tidb does not support full outer join